python/paddle/fluid/tests/unittests/op_test.py

```diff
@@ -1263,3 +1284,121 @@ def _get_gradient(self,
         return list(
             map(np.array,
                 executor.run(prog, feed_dict, fetch_list, return_numpy=False)))
+
+
+'''
+The op test with int8 precision should inherit OpTestInt8.
+'''
+
+
+class OpTestInt8(OpTestBase):
+    pass
+
+
+'''
+The op test with float16 precision should inherit OpTestFp16,
+which requires the test to call check_grad. 
+'''
+
+
+class OpTestFp16(OpTestBase):
+    def check_output(self,
+                     atol=1e-5,
+                     no_check_set=None,
+                     equal_nan=False,
+                     check_dygraph=True,
+                     inplace_atol=None,
+                     check_compile_vs_runtime=False):
+        self.__class__.op_type = self.op_type
+        OpTestBase.check_output(self, atol, no_check_set, equal_nan,
+                                check_dygraph, inplace_atol,
+                                check_compile_vs_runtime)
+
+    def check_grad(self,
+                   inputs_to_check,
+                   output_names,
+                   no_grad_set=None,
+                   numeric_grad_delta=0.005,
+                   in_place=False,
+                   max_relative_error=0.005,
+                   user_defined_grads=None,
+                   check_dygraph=True):
+        self.infer_dtype_from_inputs_outputs(self.inputs, self.outputs)
+        assert self.dtype == np.float16, "The dtype of this test should be float16."
+
+        self.__class__.op_type = self.op_type
+        self.__class__.exist_check_grad = True
+
+        OpTestBase.check_grad(self, inputs_to_check, output_names, no_grad_set,
+                              numeric_grad_delta, in_place, max_relative_error,
+                              user_defined_grads, check_dygraph)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Restore random seeds"""
+        np.random.set_state(cls._np_rand_state)
+        random.setstate(cls._py_rand_state)
+
+        if cls.__name__ not in op_white_list.NO_NEED_FP16_CHECK_GRAD_CASES \
+            and not hasattr(cls, "exist_check_grad") \
+            and cls.op_type not in op_white_list.NO_FP16_CHECK_GRAD_OP_LIST:
+            raise AssertionError("This test of %s op needs check_grad." %
+                                 cls.op_type)
+
+
+'''
+The op test with float32/64 precision should inherit OpTestFp16,
```

Comment:
should inherit OpTestFp16-》should inherit OpTest？

Reply:
Done